### ISSUE TITLE
Create new error handling macro for ESMF calls with 2 error codes

### ISSUE THREAD

**Issue by tclune** [GEOS-ESM/MAPL#1558]
Several ESMF procedures return 2 (optional) error codes.  One for ESMF internal and one for user code that is called within.  Usually the former is `rc` and the latter is `userRc`.     Currently MAPL uses a combination of two macros to check these cases, but it happens just often enough that a new macro that checks both would possibly be useful.   The counter argument would be if we were to start returning different messages depending on which failed.

Suggested names for new macro are `_VERIFY_ESMF` or `_VERIFY_USER`.

**Comment by mathomp4** [comment]
Obviously marking long-term.

Query for @tclune I'm guessing the `VERIFY_ESMF_` we have in [`NUOPC_ErrLog.h`](https://github.com/GEOS-ESM/MAPL/blob/main/include/NUOPC_ErrLog.h) is not the same as what this wants?

**Comment by tclune** [comment]
Correct.  Though important to keep in mind these others exist already.

**Comment by tclune** [comment]
Turns out this macro already exists, but ... recent experience shows that it is important to check the ESMF RC code first.  This is because certain failures will leave the user RC unassigned which leads to confusing results.    

Not sure what the best solution is, but reversing the checks in this PR.  (And exercising in MAPL3)
